source/app/service-providers/translations/index.ts

```diff
@@ -107,18 +107,23 @@ export default class TranslationProvider extends ProviderContract {
 
     // return file
     let response
-    try {
-      response = await got(TRANSLATION_API_URL, { method: 'GET' })
-    } catch (err: any) {
-      // Not critical.
-      this._logger.warning(`[Translation Provider] Could not update translations: ${String(err.code)}`, err)
-      return
+    const checkForTranslationUpdates: boolean = this._config.get('system.checkForTranslationUpdates')
```

Comment:
This long `if`-statement makes this hard to read; is it possible to just check for `if (shouldNotCheck)` to return early here? As far as I can see, the rest of the code just pertains to downloading updates.

Also, it might make sense to factor that logic out into its own function, right? It looks like a remnant from back when I didn't properly organize my code …

Reply:
> This long `if`-statement makes this hard to read; is it possible to just check for `if (shouldNotCheck)` to return early here? As far as I can see, the rest of the code just pertains to downloading updates.
> 
> Also, it might make sense to factor that logic out into its own function, right? It looks like a remnant from back when I didn't properly organize my code …

I updated the logic and moved the downloading process to update(), and keep the check process in boot(). Is it better now?

Reply:
Yes, wonderful this way!